packages/nixos/peerpods.nix

```diff
@@ -97,6 +97,20 @@ in
         ExecStop = "${pkgs.iproute2}/bin/ip netns del %I";
       };
     };
+    # Contrary to bare-metal, a peer pod needs regular network access and DNS. The default setup
+    # with dhcpcd and resolvconf does not play well with the immutable /etc, so we use the full
+    # systemd stack instead.
+    networking.dhcpcd.enable = false;
+    systemd.network.enable = true;
+    networking.useNetworkd = true;
+    services.resolved.enable = true;
+
+    # The /etc/machine-id should be populated by systemd, but the immutable /etc seems to prevent
+    # that. However, systemd-networkd requires this file to exist, so we add an empty file which
+    # will then be bind-mounted over by systemd. Let's hope no important services are gated on
+    # first-boot.
+    # https://www.man7.org/linux//man-pages/man5/machine-id.5.html#FIRST_BOOT_SEMANTICS
+    environment.etc."machine-id".text = "";
```

Comment:
Since we're immutable, essentially every boot is a "first boot". So wouldn't it be more correct to populate this file with "uninitialized"?